plugins/woocommerce/client/blocks/assets/js/blocks/add-to-cart-with-options/grouped-product-selector/product-item-template/edit.tsx

```diff
@@ -87,32 +88,67 @@ export default function ProductItemTemplateEdit(
 	} );
 
 	const { product } = useProductDataContext();
+	const [ groupedProducts, setGroupedProducts ] = useState<
+		ProductResponseItem[] | null
+	>( null );
 
-	const { products } = useSelect(
-		( select ) => {
-			const postType = 'product';
-			const { getEntityRecords } = select( coreStore );
+	useEffect( () => {
+		const fetchGroupedProducts = async (
+			productContext: ProductResponseItem[]
+		) => {
+			if ( ! productContext || productContext.length === 0 ) {
+				return;
+			}
+
+			let query: Record< string, unknown > = { per_page: 3 };
```

Comment:
I know this was like that in `trunk`, but wondering if we could make this `per_page` match `productContext[ 0 ].grouped_products` when it exists? Maybe setting a limit like 10 or 30 so it doesn't impact the editor's performance too much.

Reply:
Nice, good shout. I've set `per_page` to the number of grouped products found in the provided context, if grouped products are found, otherwise it still defaults to 3: https://github.com/woocommerce/woocommerce/pull/57690/commits/3c88f5534d558da8c29ea59b7cc69848c42ae1d3. Does this sound better, or would it be better to increase the fallback to something like 10?

Reply:
Thanks! I think it should be fine. We can iterate on it later if we receive reports of performance issues on grouped products with many children.